docs/meshstack.customer-group-sync.md

```diff
@@ -4,3 +4,206 @@ title: meshCustomer Group Synchronisation
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let ConnectionConfiguration =
+    {-
+      url: The LDAP connection URL which would be of the form ldap://example.com:389
+      base:
+          The base from which searches should be performed, which would be of the form "dc=meshcloud,dc=io"
+      username: The username of the user who can connect to the LDAP source to perform searches
+      password: The password of the user above
+    -}
+      { url : Text, base : Text, username : Text, password : Secret }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The configuration related to the collection of data via LDAP is as follows.
+
+<!--snippet:mesh.identityconnector.ldap.collector#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+  pageSize: The size of a single page that will be returned as a result of an LDAP search
+  group/person:
+    a configuration of type LDAPFilter which will contain the information needed to search the LDAP source
+    for users and groups.
+-}
+  let CollectorConfiguration =
+        { pageSize : Natural, group : LdapFilter, person : LdapFilter }
+
+  in  CollectorConfiguration
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The LdapFilter mentioned above has the following form.
+
+<!--snippet:mesh.identityconnector.ldap.filter#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+{-
+    attributes:
+        A comma separated list of attributes that should be returned per LDAP entry.
+        For example, "cn, uniqueMember, description"
+    base: The base from which the query should be performed. For example, "ou=groups"
+    filter: A filter that follows the LDAP search filter format. For example, "'(cn=mc*)'"
+-}
+  let LdapFilter = { attributes : Text, base : Text, filter : Text }
+
+  in  LdapFilter
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+
+The following configuration describes how the collected data should be transformed into meshObjects.
+<!--snippet:mesh.identityconnector.ldap.transformation#type-->
+
+
+<!--DOCUSAURUS_CODE_TABS-->
+<!--Dhall Type-->
+```dhall
+let TransformationConfiguration =
+    {-
+    mesh-object:
+      The type of meshObject this configuration is about. Can be one of
+      MeshUser, MeshCustomerUserGroup or MeshCustomerGroupBinding
+    fields:
+      A list of configuration objects specifying which LDAP attribute gets mapped to which
+      mesh object attribute, and whether any transformations should be done on the attribute before
+      assigning it to the meshObject field.
+    tags: Same as fields above, but for tags on the specified meshObject.
+    -}
+      { mesh-object : Text, fields : List Field, tags : List Field }
+```
+<!--END_DOCUSAURUS_CODE_TABS-->
+
+The Field configuration mentioned above has two possible forms. A basic form known as `BaseField` and a Field that performs
+transformations on the matched attribute, called the `RegexField`. The `RegexField` is an extension of the `BaseField`.
```

Comment:
```suggestion
transformations on the matched attribute, called the `RegexField`.
```
d: think that info is not required for the reader.

Reply:
When configuring a RegexField, they would also have to include the properties of the BaseField. So I think the information is useful. They can see that in the dhall model below as well.